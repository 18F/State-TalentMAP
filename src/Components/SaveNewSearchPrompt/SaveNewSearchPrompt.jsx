import React from 'react';
import PropTypes from 'prop-types';
import { inputKeyUp } from '../../utilities';

const SaveNewSearchPrompt = ({ toggleInput, newSavedSearchSuccess }) => (
  <div className="usa-grid-full">
    <div className="usa-grid-full">
      <a
        className="save-search-link"
        tabIndex="0"
        role="link"
        onClick={toggleInput}
        onKeyUp={(e) => { if (inputKeyUp(e)) { toggleInput(); } }}
      >
        Save this search.
      </a>
      &nbsp;You will be able to come back to these results later.
    </div>
    {
      newSavedSearchSuccess &&
      <div className="usa-grid-full">
        {newSavedSearchSuccess}
      </div>
    }
  </div>
);

SaveNewSearchPrompt.propTypes = {
  toggleInput: PropTypes.func.isRequired,
  newSavedSearchSuccess: PropTypes.string,
};

SaveNewSearchPrompt.defaultProps = {
  newSavedSearchSuccess: false,
};

export default SaveNewSearchPrompt;
