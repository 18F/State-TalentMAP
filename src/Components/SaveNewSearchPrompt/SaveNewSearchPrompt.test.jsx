import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import SaveNewSearchPrompt from './SaveNewSearchPrompt';

describe('SaveNewSearchPromptComponent', () => {
  let wrapper = null;

  it('is defined', () => {
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={() => {}}
        newSavedSearchSuccess="success"
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can handle receiving currently selected saved searches', () => {
    const savedSearch = { id: 1, name: 'test' };
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={() => {}}
        newSavedSearchSuccess="success"
        currentSavedSearch={savedSearch}
      />,
    );
    expect(wrapper.find(savedSearch.name)).toBeDefined();
    expect(wrapper.find('Edit')).toBeDefined();
  });

  it('can respond to an enter key on the link', () => {
    const spy = sinon.spy();
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={spy}
        newSavedSearchSuccess="success"
      />,
    );
    wrapper.find('a').simulate('keyUp', { keyCode: 13 });
    sinon.assert.calledOnce(spy);
    // should only respond to 13
    wrapper.find('a').simulate('keyUp', { keyCode: 14 });
    // therefore, it should still have only been called once
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot', () => {
    wrapper = shallow(
      <SaveNewSearchPrompt
        toggleInput={() => {}}
        newSavedSearchSuccess="success"
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
