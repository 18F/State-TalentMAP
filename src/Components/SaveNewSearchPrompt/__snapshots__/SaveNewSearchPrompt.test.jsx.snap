// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveNewSearchPromptComponent matches snapshot 1`] = `
<div
  className="usa-grid-full"
>
  <div
    className="usa-grid-full"
  >
    <a
      className="save-search-link"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="link"
      tabIndex="0"
    >
      Save this search.
    </a>
    Â 
    You will be able to come back to these results later.
  </div>
  <div
    className="usa-grid-full"
  >
    success
  </div>
</div>
`;
